docs/application/native/guides/machine-learning/machine-learning-inference.md

```diff
@@ -0,0 +1,350 @@
+
+  ```
+
+  First, getting `ml_pipeline_switch_h` is required. The name of switch in this pipeline is `ins`
+
+  ```bash
+    ml_pipeline_h handle;
+    ml_pipeline_switch_h switchhandle;
+
+    status = ml_pipeline_switch_get_handle (handle, "ins", &type, &switchhandle);
+  ```
+
+  Now, the switch is controllable with getting handle `ml_pipeline_switch_h`.
+
+  ```bash
+    ml_pipeline_h handle;
+    ml_pipeline_switch_h switchhandle;
+    ml_pipeline_switch_e type;
+
+    status = ml_pipeline_construct (pipeline, NULL, NULL, &handle);
+    status = ml_pipeline_switch_get_handle (handle, "ins", &type, &switchhandle);
+  ```
+
+  Before destroy the pipeline, `ml_pipeline_switch_h` has to be released.
+
+  ```bash
+    status = ml_pipeline_switch_release_handle (switchhandle);
+  ```
+
+  Switch can be located at the end of the pipeline like below image.
+
+  ![output-selector](./media/output-selector.png)
+
+  ```bash
+    gchar * pipeline = g_strdup ("videotestsrc is-live=true ! videoconvert ! tensor_converter ! output-selector name=outs "
+        "outs.src_0 ! tensor_sink name=sink0 async=false "
+        "outs.src_1 ! tensor_sink name=sink1 async=false");
+  ```
+
+### Pipeline States <a name="pipelinestate"></a>
+  You can get the information about the states at [Gstreamer guide](https://gstreamer.freedesktop.org/documentation/plugin-development/basics/states.html).
+
+## Tensors Info <a name="tensorsinfo"></a>
+
+This is a structure which contains the information of the Tensors. With below apis, `tensor info` can be managed.
+
+- Create & Destroy
+  ```bash
+    ml_tensors_info_h info;
+    status = ml_tensors_info_create (&info);
+    status = ml_tensors_info_destroy (info);
+  ```
+
+- Set APIs
+  ```bash
+    /* Set how many tensors exist */
+    status = ml_tensors_info_set_count (info, 1);
+
+    /* Set the type of the tensor_0 as UINT8 */
+    status = ml_tensors_info_set_tensor_type (info, 0, ML_TENSOR_TYPE_UINT8);
+
+    /* Set the dimension of the tensor_0 as in_dim */
+    status = ml_tensors_info_set_tensor_dimension (info, 0, in_dim);
+
+    /* Set the name of the tensor_0 as "tensor-name-test" */
+    status = ml_tensors_info_set_tensor_name (info, 0, "tensor-name-test");
+  ```
+
+- Get APIs
+  ```bash
+    /* Get how many tensors exist */
+    status = ml_tensors_info_get_count (info, &num);
+
+    /* Get the type of the tensor_0 */
+    status = ml_tensors_info_get_tensor_type (info, 0, &out_type);
+
+    /* Get the dimension of the tensor_0 */
+    status = ml_tensors_info_get_tensor_dimension (info, 0, in_dim);
+
+    /* Get the name of the tensor_0 */
+    status = ml_tensors_info_get_tensor_name (info, 0, &out_name);
+
+    /* Getaks  the size of the tensor_0 */
+    status = ml_tensors_info_get_tensor_size (info, 0, &data_size);
+  ```
+
+## Tensors Data <a name="tensorsdata"></a>
+
+This is a structure which contains the raw data of tensors. With below apis, `tensor data` can be managed.
```

Comment:
Please rephrase to:
Tensors data is a structure that contains the raw data of tensors. The `tensor data` can be managed using the following APIs: